backend/internal/ou/hybrid_store.go

```diff
@@ -0,0 +1,465 @@
+		return 0, fmt.Errorf("failed to get file store list: %w", err)
+	}
+
+	dbOUs, err := h.dbStore.GetOrganizationUnitList(9999, 0)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get database store list: %w", err)
+	}
+
+	// Create set of file-based IDs
+	fileIDs := make(map[string]bool)
+	for _, ou := range fileOUs {
+		fileIDs[ou.ID] = true
+	}
+
+	// Count unique DB OUs (exclude duplicates)
+	uniqueDBCount := 0
+	for _, ou := range dbOUs {
+		if !fileIDs[ou.ID] {
+			uniqueDBCount++
+		}
+	}
+
+	return fileCount + uniqueDBCount, nil
+}
+
+// GetOrganizationUnitList retrieves root organization units from both stores with pagination.
+// File-based OUs are returned first, followed by database OUs.
+func (h *ouHybridStore) GetOrganizationUnitList(limit, offset int) ([]OrganizationUnitBasic, error) {
+	// Get all from both stores
+	fileOUs, err := h.fileStore.GetOrganizationUnitList(9999, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get file store list: %w", err)
+	}
+
+	dbOUs, err := h.dbStore.GetOrganizationUnitList(9999, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get database store list: %w", err)
+	}
+
+	// Merge: file-based first, then DB (avoid duplicates by ID)
+	seen := make(map[string]bool)
+	var merged []OrganizationUnitBasic
+
+	for _, ou := range fileOUs {
+		merged = append(merged, ou)
+		seen[ou.ID] = true
+	}
+
+	for _, ou := range dbOUs {
+		if !seen[ou.ID] {
+			merged = append(merged, ou)
+		}
+	}
+
+	// Apply pagination to merged results
+	start := offset
+	if start > len(merged) {
+		return []OrganizationUnitBasic{}, nil
+	}
+
+	end := start + limit
+	if end > len(merged) {
+		end = len(merged)
+	}
+
+	return merged[start:end], nil
+}
+
+// CreateOrganizationUnit creates a new organization unit in the database store only.
+// It checks for conflicts with file-based OUs before creating.
+func (h *ouHybridStore) CreateOrganizationUnit(ou OrganizationUnit) error {
+	// Check if ID exists in file store (immutable)
+	_, err := h.fileStore.GetOrganizationUnit(ou.ID)
+	if err == nil {
+		return errors.New("cannot create organization unit: ID conflicts with immutable file-based OU")
+	}
+	if err != ErrOrganizationUnitNotFound {
+		return fmt.Errorf("error checking file store: %w", err)
+	}
+
+	// Check handle conflict in file store
+	hasConflict, err := h.fileStore.CheckOrganizationUnitHandleConflict(ou.Handle, ou.Parent)
+	if err != nil {
+		return fmt.Errorf("error checking handle conflict in file store: %w", err)
+	}
+	if hasConflict {
+		return errors.New("cannot create organization unit: handle conflicts with immutable file-based OU")
+	}
+
+	// Check name conflict in file store
+	hasConflict, err = h.fileStore.CheckOrganizationUnitNameConflict(ou.Name, ou.Parent)
+	if err != nil {
+		return fmt.Errorf("error checking name conflict in file store: %w", err)
+	}
+	if hasConflict {
+		return errors.New("cannot create organization unit: name conflicts with immutable file-based OU")
+	}
```

Comment:
shouldn't this part of the service layer?

Service layer will call `hybridStore.CheckOrganizationUnitNameConflict` and hybridStore will check from both stores